aws-cpp-sdk-core/source/utils/Directory.cpp

```diff
@@ -63,6 +64,36 @@ namespace Aws
 			return ss.str();
 		}
 
+        Aws::String GetDirectoryName(const char* fileName)
+        {
+            Aws::String pathName = Aws::String(fileName);
+
+            if (pathName.empty()) return ".";
+
+            const auto lastPathDelim = std::find(pathName.rbegin(),
+                                                 pathName.rend(),
+                                                 PATH_DELIM);
+
+            if (lastPathDelim == pathName.rend())
+            {
+                // No path delimiters found
+                return ".";
+            }
+            else if (pathName.front() == PATH_DELIM)
+            {
+                // Path delimiter is first char
+                return Aws::String("") + PATH_DELIM;
+            }
+            else if (pathName.back() == PATH_DELIM)
+            {
+                // Path delimiter is last char
+                return GetDirectoryName(
+                        Aws::String(pathName.begin(), pathName.end() - 1).c_str());
```

Comment:
It's used in order to get the path from filenames like `d/f/`, where the `/` will be last character and the `d/f` string will be processed.